runtime/runtime/src/ext.rs

```diff
@@ -180,4 +184,43 @@ impl<'a> External for RuntimeExt<'a> {
             .validator_total_stake(self.epoch_id, self.prev_block_hash)
             .map_err(|e| ExternalError::ValidatorError(e).into())
     }
+
+    fn get_block_hash_by_height(&self, height_number: BlockHeight) -> (usize, Option<CryptoHash>) {
+        let chain_store = self.chain_store;
+        let tip = chain_store.header_head().expect("handle this properly");
+        // requested height isn't available
+        if tip.height < height_number {
+            return (0, None);
+        }
+
+        if tip.height == height_number {
+            return (0, Some(tip.last_block_hash));
+        }
+        let mut previous_block_hash = tip.prev_block_hash;
+        if previous_block_hash == CryptoHash::default() {
+            // this indicates it's the genesis block
+            return (0, None);
+        }
+        // iterate up to 255 times more
+        for iteration in 0..255 {
+            let block_header =
+                chain_store.get_block_header(&previous_block_hash).expect("handle this properly");
+
+            match block_header.height().cmp(&height_number) {
+                std::cmp::Ordering::Less => {
+                    // User requested block height which does not exist, i.e. a block height
+                    // has been skipped
+                    return (iteration, None);
```

Comment:
I think this should be iteration + 1? (same a few lines further down)

Reply:
yes!